### PR TITLE
fix: Avoid FlowProvider reload on trigger launch

### PR DESCRIPTION
**Pull request by doubleface**
With `harvest.inappconnectors.enabled` flag enabled, when launching a trigger,
there is a flicker which is caused by the FlowProvider component reload.

To fix it, we change the component tree structure to avoid the rerender
of the AccountModal component.

This fix has been tested in home application with and without the flag
enabled and also in the `Mes Papier` application.

- fix: Avoid FlowProvider reload on trigger launch
- fix: Remove a check in AccountModal unit test
- fix: Make withRouter unit tests more independant

### CODE REVIEW

File: packages/cozy-harvest-lib/src/components/KonnectorConfiguration/KonnectorAccountTabs.jsx [hunk 1]
```diff
@@ -191,27 +185,4 @@ const DumbKonnectorAccountTabs = props => {
...
-  konnector: PropTypes.object.isRequired,
-  initialTrigger: PropTypes.object.isRequired,
-  account: PropTypes.object.isRequired,
-  onAccountDeleted: PropTypes.func.isRequired,
-  addAccount: PropTypes.func.isRequired,
-  /** @type {string} Can be used to force the initial tab */
-  initialActiveTab: PropTypes.oneOf(['configuration', 'data']),
-  intentsApi: intentsApiProptype,
-  innerAccountModalOverrides: innerAccountModalOverridesProptype
-}
```
Comment (Crash--) [inline packages/cozy-harvest-lib/src/components/KonnectorConfiguration/KonnectorAccountTabs.jsx:215]: why are we deleting the propTypes? Can we add them to `KonnectorAccountTabs` components?

File: packages/cozy-harvest-lib/src/components/AccountModal.jsx [hunk 2]
```diff
@@ -121,29 +123,39 @@ export class AccountModal extends Component {
       showAccountSelection,
       showNewAccountButton,
       intentsApi,
-      innerAccountModalOverrides
+      innerAccountModalOverrides,
+      component
```
Comment (Crash--) [inline packages/cozy-harvest-lib/src/components/AccountModal.jsx:127]: propTypes to update

File: packages/cozy-harvest-lib/src/components/KonnectorConfiguration/KonnectorAccountWrapper.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  konnector: PropTypes.object.isRequired,
+  initialTrigger: PropTypes.object.isRequired,
+  account: PropTypes.object.isRequired,
+  onAccountDeleted: PropTypes.func.isRequired,
+  addAccount: PropTypes.func.isRequired,
+  /** @type {string} Can be used to force the initial tab */
+  initialActiveTab: PropTypes.oneOf(['configuration', 'data']),
+  intentsApi: intentsApiProptype,
+  innerAccountModalOverrides: innerAccountModalOverridesProptype
+}
```
Comment (Crash--) [inline packages/cozy-harvest-lib/src/components/KonnectorConfiguration/KonnectorAccountWrapper.jsx:67]: component to add?

File: packages/cozy-harvest-lib/src/components/KonnectorConfiguration/KonnectorAccountWrapper.jsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      break
+
+    case 'ConfigurationTab':
+      Component = ConfigurationTab
+      break
+
+    default:
+      Component = DataTab
+      break
+  }
```
Comment (Crash--) [inline packages/cozy-harvest-lib/src/components/KonnectorConfiguration/KonnectorAccountWrapper.jsx:33]: I don't like this kind of API. Why passing a string to translate it in component later? Why not passing directly the component? 🙏

File: packages/cozy-harvest-lib/src/components/Routes/RoutesV4.jsx [hunk 5]
```diff
@@ -62,48 +57,30 @@ const RoutesV4 = ({
...
-                <AccountModalContentWrapper>
-                  <DataTab
-                    konnectorRoot={konnectorRoot}
-                    konnector={konnectorWithTriggers}
-                    showNewAccountButton={!konnectorWithTriggers.clientSide}
-                    onDismiss={onDismiss}
-                  />
-                </AccountModalContentWrapper>
-              </AccountModalWithoutTabs>
+                component="DataTab"
```
Comment (Crash--) [inline packages/cozy-harvest-lib/src/components/Routes/RoutesV4.jsx:67]: exactly what I don't want. https://github.com/cozy/cozy-libs/pull/2139/commits/571ba07e144d1a18c6fbcd2afc6cb42cbaa711bc#r1183285535

File: packages/cozy-harvest-lib/src/components/Routes/RoutesV6.jsx [hunk 6]
```diff
@@ -57,49 +45,44 @@ const RoutesV6 = ({
...
+                  />
+                )}
+              </HarvestParamsWrapper>
             }
           />
-        </Route>
+        </>
       ) : (
         <Route
           path="accounts/:accountId"
```
Comment (Crash--) [inline packages/cozy-harvest-lib/src/components/Routes/RoutesV6.jsx:88]: @JF-Cozy / @Merkur39 Can you review this change please?

File: packages/cozy-harvest-lib/src/components/AccountModal.spec.jsx [hunk 7]
```diff
@@ -99,10 +99,6 @@ describe('AccountModal', () => {
 
       component.find('AccountSelectBox').prop('onCreate')()
       expect(mockHistoryPush).toHaveBeenCalledWith('/new')
-
-      const accountTabs = component.find('KonnectorAccountTabs')
-      accountTabs.prop('addAccount')()
-      expect(mockHistoryPush).toHaveBeenCalledWith('/new')
```
Comment (Crash--) [inline packages/cozy-harvest-lib/src/components/AccountModal.spec.jsx:105]: ok but should we add it in another test?

File: packages/cozy-harvest-lib/src/components/Routes/RoutesV6.jsx (hunk 6 above)
Comment (JF-Cozy) [reply packages/cozy-harvest-lib/src/components/Routes/RoutesV6.jsx:88]: sur la route ça me semble ok, sur les enfants rendus, après discussion avec @Merkur39 je lui laisse la main pour répondre (il a plus le sujet en tête que moi)

File: packages/cozy-harvest-lib/src/components/AccountModal.jsx [hunk 8]
```diff
@@ -121,29 +123,39 @@ export class AccountModal extends Component {
...
-            <Typography>{models.account.getAccountName(account)}</Typography>
-          )}
-        </KonnectorModalHeader>
+        {flag('harvest.inappconnectors.enabled') ? (
+          <AccountModalHeader
+            konnector={konnector}
+            account={account}
+            accountsAndTriggers={accountsAndTriggers}
+          />
+        ) : (
```
Comment (JF-Cozy) [inline packages/cozy-harvest-lib/src/components/AccountModal.jsx:139]: pour plus de lisibilité on pourrait écrire

```
if (flag('harvest.inappconnectors.enabled') {
  return ()
}

return ()
```

File: packages/cozy-harvest-lib/src/components/Routes/RoutesV6.jsx (hunk 6 above)
Comment (Merkur39) [reply packages/cozy-harvest-lib/src/components/Routes/RoutesV6.jsx:88]: Ca me parait OK, n'utilisant plus un parent commun (`AccountModalWithoutTabs`), dans ce cas présent autant simplifier

File: packages/cozy-harvest-lib/src/components/AccountModal.spec.jsx (hunk 7 above)
Comment (doubleface) [reply packages/cozy-harvest-lib/src/components/AccountModal.spec.jsx:105]: My understanding of this test was wrong. Now the test is kept and I removed this commit.

**Comment by doubleface** [comment]
@Merkur39 @Crash-- J'ai mis cette PR à jour

**Review by Crash-- — APPROVED** [review]

**Pull request merged**